client/db/bolt/db.go

```diff
@@ -1629,6 +1650,70 @@ func loadMatchBucket(mBkt *bbolt.Bucket, excludeCancels bool) (*dexdb.MetaMatch,
 	}, nil
 }
 
+// errFound is used to break ForEach early when a match is found.
+var errFound = errors.New("found")
+
+// hasTradeMatch checks if an order has any non-cancel trade matches.
+// A trade match has a non-empty Address (not a cancel match for maker)
+// and has InitSig (not a cancel match for taker).
+// Returns true immediately upon finding the first trade match (early exit).
+func (db *BoltDB) hasTradeMatch(oid order.OrderID) bool {
+	found := false
+	db.matchesView(func(mb, archivedMB *bbolt.Bucket) error {
+		for _, bucket := range []*bbolt.Bucket{mb, archivedMB} {
+			if bucket == nil {
+				continue
+			}
+			err := bucket.ForEach(func(k, _ []byte) error {
+				mBkt := bucket.Bucket(k)
+				if mBkt == nil {
+					return nil
+				}
+				// Check if match belongs to this order
+				if !bytes.Equal(mBkt.Get(orderIDKey), oid[:]) {
+					return nil
+				}
+
+				// Decode match (lightweight - only UserMatch)
+				matchB := mBkt.Get(matchKey)
+				if matchB == nil {
+					return nil
+				}
+				match, _, err := order.DecodeMatch(matchB)
+				if err != nil {
+					return nil // Skip on error
+				}
+
+				// Cancel match for maker has empty Address
+				if match.Address == "" {
+					return nil
+				}
+
+				// Cancel match for taker has no InitSig and status MatchComplete
+				// Only decode proof when necessary
+				if match.Status == order.MatchComplete {
+					proofB := mBkt.Get(proofKey)
+					if len(proofB) > 0 {
+						proof, _, err := dexdb.DecodeMatchProof(proofB)
+						if err == nil && len(proof.Auth.InitSig) == 0 {
+							return nil // Cancel match for taker
+						}
+					}
+				}
+
+				// Found a trade match - early exit!
+				found = true
+				return errFound
+			})
+			if err == errFound {
+				return errFound // Propagate to outer loop
+			}
+		}
+		return nil
```

Comment:
Could you use https://pkg.go.dev/errors#Is please?

Also, if there is any other error, please log at the end. So please inspect the error returned from matchesView. Currently it is ignored.